### ISSUE TITLE
No Error Reported when Closing Issue Failed due to Underlying Dependency

### ISSUE THREAD

**Issue by CasperHK** [go-gitea/gitea#18236]
### Gitea Version

1.16.0

### Operating System

Ubuntu

### How are you running Gitea?

I host it by myself in a VM inside intranet.

### Database

MySQL

### Can you reproduce the bug on the Gitea demo site?

No

### Description

I found an error when closing an issue. When I click on the **Close** button in the bottom comment section, the website get redirected to the same page but the issue is not yet closed. I cannot reproduce the same error on other issue post on the same site. I consider it may be a bug in Gitea.

### Current Action
Currently, I just close the related post manually by setting the `is_closed` flag to `1` in the database, which means closed. However, the problem is that the person who closed the post does not record in the history.

Someone encountered the same problem before? And, how can you fix it? Please help.

Thank you.

**Comment by zeripath** [comment]
Why do we always have to repeat this: 

Please read: https://docs.gitea.io/en-us/logging-configuration/#debugging-problems

There will almost certainly be something helpful to help further explain what you're seeing in the logs. Could you please give us some logs.

However, in this case we're very likely to need `LOG_SQL=true` here too.

So please could you set your logging to:

```ini
[database]
LOG_SQL = true ; Normally we don't want or need SQL logging for debug logs however, in this case it's likely to be helpful.

...

[log]
MODE = console
LEVEL = debug ; please set the level to debug when we are debugging a problem
ROUTER = console
COLORIZE = false ; this can be true if you can strip out the ansi coloring
```

**Comment by GaLaXy102** [comment]
I was able to reproduce this issue on my instance when trying to close an issue with dependencies. It appears you can't close an issue when there are other issues blocking it. There is no error message whatsoever, just nothing happens. I have the Logs attached.

It would be great if Gitea would send an alert stating the dependency.
[gitea.log](https://github.com/go-gitea/gitea/files/7874764/gitea.log)

**Comment by zeripath** [comment]
It think this may be  related to #17643

**Comment by zeripath** [comment]
There should be an error reported - but it's not happening.

**Comment by zeripath** [comment]
Yup this is a regression due to #17643 - we should be handling the dependency error in the callers.

**Comment by zeripath** [comment]
Duplicate of #18223

**Comment by CasperHK** [comment]
Due to no error message shown to the user, user may directly thinking of whether it is a bug. This means if the dependence not yet closed, then the target issue will not be able to be closed too. Then, the error message should be quite important.

**Comment by zeripath** [comment]
This is already fixed in 1.16+ by #18325. If you're seeing this problem in 1.16 you need to open a new issue and give us full details. We will not backport this to 1.15.

Commenting on **duplicate** issues  which have been closed over a month ago and where the original issue already has been fixed by a PR is not helpful.
